zhihu-remove-login-modal.js

```diff
@@ -17,22 +17,33 @@
     window.addEventListener('keydown', e => e.keyCode === 27 && e.stopImmediatePropagation(), true)
 
     // watch madal
-    let disabled = false
+    let modalDisabled = false
+    let footDialog = null
 
     const body = document.body
     const html = document.documentElement
 
     const getModal = () => document.querySelector('.signFlowModal')
 
+    // const getFootDialog = () => document.querySelector('img[src="https://static.zhihu.com/heifetz/assets/liukanshan-peek.a71ecf3e.png"]')
+    const getFootDialog = () => Array.from(document.querySelectorAll('button')).filter(item => item.innerText.match(/立即登录/))[0]
```

Comment:
I am not sure if this regex will be compiled every time, but it shouldn't be a bad idea to extract a variable

Reply:
Sorry for not expressing clearly, what I mean is that we should have the regex as a variable instead of creating it repeatedly in a loop:

```js
const RE_BTN = /立即登录/
const getFootDialog = () => Array.from(document.querySelectorAll('button')).filter(RE_BTN.test.bind(RE_BTN))[0]
```

Reply:
you are right！